File: typescript-playwright/src/api/AdminCommentApi.ts
import type { APIRequestContext, APIResponse } from "@playwright/test";
import type { CommentStatus } from "../utils/types";
import { API_URL } from "../utils/constants";
import { mergeHeaders } from "../utils/headers";

export class AdminCommentApi {
  private readonly request: APIRequestContext;
  private readonly authHeaders: Record<string, string>;

  constructor(
    request: APIRequestContext,
    authHeaders: Record<string, string>,
  ) {
    this.request = request;
    this.authHeaders = authHeaders;
  }

  private get headers(): Record<string, string> {
    return mergeHeaders(this.authHeaders);
  }

  async getComments(page = 0, size = 20): Promise<APIResponse> {
    return this.request.get(`${API_URL}/api/admin/comments`, {
      params: { page, size },
      headers: this.headers,
    });
  }

  async getStats(): Promise<APIResponse> {
    return this.request.get(`${API_URL}/api/admin/comments/stats`, {
      headers: this.headers,
    });
  }

  async moderate(
    commentId: number,
    status: CommentStatus,
    rejectionReason?: string,
  ): Promise<APIResponse> {
    return this.request.put(
      `${API_URL}/api/admin/comments/${commentId}/moderate`,
      {
        data: { status, ...(rejectionReason ? { rejectionReason } : {}) },
        headers: this.headers,
      },
    );
  }

  async deleteComment(commentId: number): Promise<APIResponse> {
    return this.request.delete(`${API_URL}/api/admin/comments/${commentId}`, {
      headers: this.headers,
    });
  }
}

File: python-pytest/src/api/admin_comment_api.py
from __future__ import annotations

import requests

from src.utils.constants import API_URL
from src.utils.headers import merge_headers
from src.utils.types import CommentStatus


class AdminCommentApi:
    def __init__(self, session: requests.Session, auth_headers: dict[str, str]) -> None:
        self._session = session
        self._auth_headers = auth_headers

    @property
    def _headers(self) -> dict[str, str]:
        return merge_headers(self._auth_headers)

    def get_comments(self, page: int = 0, size: int = 20) -> requests.Response:
        return self._session.get(
            f"{API_URL}/api/admin/comments",
            params={"page": page, "size": size},
            headers=self._headers,
            timeout=10,
        )

    def get_stats(self) -> requests.Response:
        return self._session.get(
            f"{API_URL}/api/admin/comments/stats",
            headers=self._headers,
            timeout=10,
        )

    def moderate(
        self, comment_id: int, status: CommentStatus, rejection_reason: str | None = None
    ) -> requests.Response:
        body: dict[str, object] = {"status": status}
        if rejection_reason:
            body["rejectionReason"] = rejection_reason
        return self._session.put(
            f"{API_URL}/api/admin/comments/{comment_id}/moderate",
            json=body,
            headers=self._headers,
            timeout=10,
        )

    def delete_comment(self, comment_id: int) -> requests.Response:
        return self._session.delete(
            f"{API_URL}/api/admin/comments/{comment_id}",
            headers=self._headers,
            timeout=10,
        )
